### PR TITLE
feat(eslint): select @o3r/eslint-config or @o3r/eslint-config-otter depending on the presence of eslintrc or not

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change

- Select `@o3r/eslint-config` or `@o3r/eslint-config-otter` depending on the presence of eslintrc or not
- Use `@o3r/eslint-config` when generating a shell sdk

### CODE REVIEW

File: packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/eslint.config.mjs.template [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import o3rConfig from '@o3r/eslint-config';
+import globals from 'globals';
+import { dirname } from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+// eslint-disable-next-line no-underscore-dangle
+const __filename = fileURLToPath(import.meta.url);
+// __dirname is not defined in ES module scope
+// eslint-disable-next-line no-underscore-dangle
+const __dirname = dirname(__filename);
```
Comment (fpaul-1A) [inline packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/eslint.config.mjs.template:10]: do we need that even for a `mjs` file ?

File: packages/@o3r/workspace/schematics/ng-add/helpers/linter.ts [hunk 2]
```diff
@@ -1,23 +1,31 @@
-import { SchematicContext } from '@angular-devkit/schematics';
+import type { JsonObject } from '@angular-devkit/core';
+import type { SchematicContext, Tree } from '@angular-devkit/schematics';
 import { askConfirmation } from '@angular/cli/src/utilities/prompt';
 
+/**
+ * It the legacy ESLint config used in the repository
```
Comment (fpaul-1A) [inline packages/@o3r/workspace/schematics/ng-add/helpers/linter.ts:6]:
```suggestion
 * If the legacy ESLint config is used in the repository
```
?
Also I would rename the function to make it more obvious that it's just a check
`useLegacyConfig` -> `isUsingLegacyConfig`

File: packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/eslint.config.mjs.template (hunk 1 above)
Comment (matthieu-crouzet) [reply packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/eslint.config.mjs.template:10]: we need that only for `mjs` files :)

**Review by fpaul-1A — APPROVED** [review]

File: packages/@o3r/workspace/schematics/ng-add/helpers/linter.ts [hunk 3]
```diff
@@ -1,23 +1,31 @@
-import { SchematicContext } from '@angular-devkit/schematics';
+import type { JsonObject } from '@angular-devkit/core';
+import type { SchematicContext, Tree } from '@angular-devkit/schematics';
 import { askConfirmation } from '@angular/cli/src/utilities/prompt';
 
+/**
+ * If the legacy ESLint config is used in the repository
+ * @param tree
+ */
+export const isUsingLegacyConfig = (tree: Tree) => tree.root.subfiles.find((file) => /\.eslintrc\.(js|cjs|yaml|yml|json)/.test(file))
```
Comment (cpaulve-1A) [inline packages/@o3r/workspace/schematics/ng-add/helpers/linter.ts:9]: Same remark as before

File: packages/@o3r/core/schematics/ng-add/utils/linter/index.ts [hunk 4]
```diff
@@ -1,19 +1,27 @@
-import { SchematicContext } from '@angular-devkit/schematics';
+import type { JsonObject } from '@angular-devkit/core';
+import { SchematicContext, type Tree } from '@angular-devkit/schematics';
 import { askConfirmation } from '@angular/cli/src/utilities/prompt';
 
+/**
+ * If the legacy ESLint config is used in the repository
+ * @param tree
+ */
+export const isUsingLegacyConfig = (tree: Tree) => tree.root.subfiles.find((file) => /\.eslintrc\.(js|cjs|yaml|yml|json)/.test(file))
```
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply packages/@o3r/core/schematics/ng-add/utils/linter/index.ts]: I reversed the condition so if both are found then we install the new one

File: packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/eslint.config.mjs.template [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          'testing/tsconfig.spec.json',
+          'tsconfigs/tsconfig.jest.json',
+          'tsconfigs/tsconfig.source.json'
+        ]
+      },
+      ecmaVersion: 12
+    }
+  },
+  {
+    name: '@o3r/globals',
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/schematics/typescript/shell/templates/base/eslint.config.mjs.template:36]: Not sure you want this rule with this name in a fresh generated SDK.
Maybe you can name it `'<%= packageName %>/globals'`

**Review by kpanot — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Pull request merged**